### PR TITLE
Remove unnecessary dependency patch on `SignatureController`

### PR DESCRIPTION
**Pull request by OGPoyraz**
## **Description**

This PR aims to remove unnecessary dependency patch in `SignatureController` (`message-manager` patch) and make `SignatureController@4.0.1` patch a bit cleaner by putting a comment (which branch patch created from) and remove unnecessary changes.

### **Why `message-manager` patch is removed?**
Since `controller-utils` is updated in to `3.4.0` in mobile repo, the patch made in the `message-manager` is became unnecessary because patched imports already been exported in `3.4.0`. 

### What changed in the `signature-controller` patch?
Same exported enums were needed in the `signature-controller`, so these patches are removed.
Also put a comment of how can we reproduce `signature-controller` patch if needed. https://github.com/MetaMask/core/tree/patch/mobile-signature-controller

## **Manual testing steps**

No functional change here.
But it's still beneficial to run regression tests around signatures.

## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change._

### **Before**

_[screenshot]_

### **After**

_[screenshot]_

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained:
  - [X] What problem this PR is solving.
  - [X] How this problem was solved.
  - [X] How reviewers can test my changes.
- [X] I’ve indicated what issue this PR is linked to: Fixes #???
- [X] I’ve included tests if applicable.
- [X] I’ve documented any added code.
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [X] I’ve properly set the pull request status:
  - [X] In case it's not yet "ready for review", I've set it to "draft".
  - [X] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by vinistevam — DISMISSED** [review]

File: patches/@metamask+signature-controller+4.0.1.patch [hunk 1]
```diff
@@ -1,7 +1,8 @@
-diff --git a/node_modules/@metamask/signature-controller/dist/.DS_Store b/node_modules/@metamask/signature-controller/dist/.DS_Store
++PATCH GENERATED FROM MetaMask/core branch: refactor/signature-controller-patch-mobile
```
Comment (matthewwalsh0) [inline patches/@metamask+signature-controller+4.0.1.patch:1]: As this is a new patch, should we use a nicer format for the core branch, maybe something like `patch/mobile-signature-controller`?

File: patches/@metamask+signature-controller+4.0.1.patch [hunk 2]
```diff
@@ -216,11 +159,11 @@ index b58b27e..24bbe8c 100644
...
...ersonalMsgs,\n      ...this.state.unapprovedTypedMessages,\n    }[messageId];\n  }\n\n  async #requestApproval(\n    msgParams: AbstractMessageParamsMetamask,\n    type: ApprovalType,\n  ): Promise<AddResult> {\n    const id = msgParams.metamaskId as string;\n    const origin = msgParams.origin || ORIGIN_METAMASK;\n\n    // We are explicitly cloning the message params here to prevent the mutation errors on development mode\n    // Because sending it through the messaging system will make the object read only\n    const clonedMsgParams = cloneDeep(msgParams);\n    return (await this.messagingSystem.call(\n      'ApprovalController:addRequest',\n      {\n        id,\n        origin,\n        type,\n        requestData: clonedMsgParams as Required<AbstractMessageParamsMetamask>,\n        expectsResult: true,\n      },\n      true,\n    )) as Promise<AddResult>;\n  }\n\n  #removeJsonData(\n    messageParams: TypedMessageParams,\n    version: string,\n  ): TypedMessageParams {\n    if (version === 'V1' || typeof messageParams.data !== 'string') {\n      return messageParams;\n    }\n\n    return {\n      ...messageParams,\n      data: JSON.parse(messageParams.data),\n    };\n  }\n}\n"]}
```
Comment (matthewwalsh0) [inline patches/@metamask+signature-controller+4.0.1.patch:168]: Can we delete all the `.map` files before generating the patch to minimise the diffs going forward.

It's currently the approach with the transaction controller patch also.

File: patches/@metamask+signature-controller+4.0.1.patch [hunk 3]
```diff
@@ -104,11 +57,11 @@ index b58b27e..24bbe8c 100644
 -            return __classPrivateFieldGet(this, _SignatureController_instances, "m", _SignatureController_newUnsignedAbstractMessage).call(this, __classPrivateFieldGet(this, _SignatureController_typedMessageManager, "f"), controller_utils_1.ApprovalType.EthSignTypedData, 'Typed Message', __classPrivateFieldGet(this, _SignatureController_instances, "m", _SignatureController_signTypedMessage).bind(this), messageParams, req, undefined, version, {
 -                parseJsonData: true,
 -            });
-+            return __classPrivateFieldGet(this, _SignatureController_instances, "m", _SignatureController_newUnsignedAbstractMessage).call(this, __classPrivateFieldGet(this, _SignatureController_typedMessageManager, "f"), ApprovalType.EthSignTypedData, 'Typed Message', __classPrivateFieldGet(this, _SignatureController_instances, "m", _SignatureController_signTypedMessage).bind(this), messageParams, req, undefined, version, signingOpts);
```
Comment (matthewwalsh0) [inline patches/@metamask+signature-controller+4.0.1.patch:107]: For these specific changes in this patch, is it possible to generate a diff in the core branch to verify them?

Did we have the previous logic in a core commit somewhere, or could we generate it for comparison sake?

File: patches/@metamask+signature-controller+4.0.1.patch (hunk 1 above)
Comment (OGPoyraz) [reply patches/@metamask+signature-controller+4.0.1.patch]: Done in https://github.com/MetaMask/metamask-mobile/pull/7355/commits/28028ba14818d1036a2adc2273eb76001c35689a

File: patches/@metamask+signature-controller+4.0.1.patch (hunk 2 above)
Comment (OGPoyraz) [reply patches/@metamask+signature-controller+4.0.1.patch]: Removed them here https://github.com/MetaMask/metamask-mobile/pull/7355/commits/3180dd46069951f8f168bf83aad5937504a13910

File: patches/@metamask+signature-controller+4.0.1.patch (hunk 3 above)
Comment (OGPoyraz) [reply patches/@metamask+signature-controller+4.0.1.patch:107]: I've temporarily placed a SC patch with version `5.1.0` and compared to this PR 
https://github.com/MetaMask/metamask-mobile/pull/7359/files
As it seen, there is only a comment diff exists.

**Comment by sleepytanya** [comment]
All regression signatures tests are passing both on iOS and Android.
Screenshot 2023-10-02 at 1 21 01 AM

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**